src/LogicEngine/RulesManager.cs

```diff
@@ -6,51 +6,73 @@
 using TinyFp;
 using TinyFp.Extensions;
 using static System.Array;
-using static TinyFp.Prelude;
 
 namespace LogicEngine;
 
 public class RulesManager<T> : IRulesManager<T> where T : new()
 {
     private readonly IRulesCatalogCompiler _catalogCompiler;
 
-    private static readonly CompiledCatalog<T> EmptyCatalog = new(Empty<Func<T, Either<string, Unit>>[]>());
-    private CompiledCatalog<T> _rulesCatalog;
-
     public RulesManager(IRulesCatalogCompiler catalogCompiler)
     {
         _catalogCompiler = catalogCompiler;
-        _rulesCatalog = EmptyCatalog;
+        _itemSatisfiesRulesWithMessage = ItemSatisfiesRulesWithMessageAlwaysUnit;
+        _itemSatisfiesRules = ItemSatisfiesRulesAlwaysTrue;
     }
 
     public RulesCatalog Catalog
     {
-        set => _rulesCatalog = _catalogCompiler.CompileCatalog<T>(value);
+        set => (_itemSatisfiesRulesWithMessage, _itemSatisfiesRules) =
+                    _catalogCompiler
+                        .CompileCatalog<T>(value)
+                        .ToOption(_ => !_.Executables.Any())
```

Comment:
You could also condense in a single `ToOption` with mapping and reduce the need to access to the `Executable` property